spec/lib/api/2.0/skus-spec.js

```diff
@@ -0,0 +1,360 @@
+            expect(sku).to.have.property('discoveryGraphOptions')
+            .that.deep.equals(input.discoveryGraphOptions);
+        });
+
+        it('should have regenerated SKUs', function () {
+            expect(workflowApiService.createAndRunGraph).to.have.been.calledOnce;
+            expect(workflowApiService.createAndRunGraph)
+                .to.have.been.calledWith({
+                    name: 'Graph.GenerateSku',
+                    options: {
+                        defaults: {
+                            nodeId: node.id
+                        }
+                    }
+                });
+        });
+
+        it('should contain the new sku in GET /skus', function () {
+            return helper.request().get('/api/2.0/skus')
+            .expect('Content-Type', /^application\/json/)
+            .expect(200, [sku]);
+        });
+
+        it('should return the same sku from GET /skus/:id', function () {
+            return helper.request().get('/api/2.0/skus/' + sku.id)
+            .expect('Content-Type', /^application\/json/)
+            .expect(200, _.merge({}, sku, { packInfo: null }) );
+        });
+
+        describe('PATCH /skus/:id', function () {
+            var updated;
+
+            beforeEach('reset createAndRunGraph stub', function () {
+                workflowApiService.createAndRunGraph.reset();
+            });
+
+            beforeEach('PATCH /skus/:id', function () {
+                sku.name = 'updated sku name';
+                return helper.request().patch('/api/2.0/skus/' + sku.id)
+                .send(sku)
+                .expect('Content-Type', /^application\/json/)
+                .expect(200)
+                .then(function (res) {
+                    updated = res.body;
+                });
+            });
+
+            it('should have an updated name', function () {
+                expect(updated).to.have.property('name').that.equals(sku.name);
+            });
+
+            it('should have the same rules', function () {
+                expect(updated).to.have.property('rules').that.deep.equals(sku.rules);
+            });
+
+            it('should have regenerated SKUs', function () {
+                expect(workflowApiService.createAndRunGraph).to.have.been.calledOnce;
+                expect(workflowApiService.createAndRunGraph)
+                    .to.have.been.calledWith({
+                        name: 'Graph.GenerateSku',
+                        options: {
+                            defaults: {
+                                nodeId: node.id
+                            }
+                        }
+                    });
+            });
+        });
+
+        describe('GET /skus/:id/nodes', function () {
+            beforeEach('assign the SKU to the node', function () {
+                var waterline = helper.injector.get('Services.Waterline');
+                return waterline.nodes.update(node.id, { sku: sku.id });
```

Comment:
! You should avoid manipulating the real database in unit-testing, you can turn to use the sinon.stub